File: c/35.c
/* 55 */
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include "euler.h"

#define N 1000000

int main(int argc, char **argv)
{
    int np;
    char *sieve;
    int *primes;
    np = prime_sieve(N, &sieve, &primes);

    char str[7];
    int cp, ncp = 0;
    int i, j, k, d, s, len;
    for (i = 0; i < np; i++)
    {
        sprintf(str, "%d", primes[i]);
        len = strlen(str);
        cp = 1;
        for (j = 1; j < len && cp; j++)
        {
            s = 0;
            for (k = 0, d = 1; k < len; k++, d *= 10)
                s += (str[len - 1 - (j + k) % len] - '0') * d;
            if (!sieve[s])
                cp = 0;
        }
        if (cp)
            ncp++;
    }
    printf("%d\n", ncp);
    free(sieve);
    free(primes);
    return 0;
}

File: py/35.py
# 55
import euler
N = 1000000
sieve = euler.prime_sieve(N)

ss = 0
for n in xrange(N):
    if sieve[n]:
        nn = n
        circular = True
        while True:
            s = str(nn)
            nn = int(s[-1] + s[:-1])
            if nn == n:
                break
            if not sieve[nn]:
                circular = False
                break
        if circular:
            ss += 1
print ss
